core/src/test/java/org/jboss/jca/core/connectionmanager/pool/PoolByCriNoTxDecrementCapacityTimedOutFIFOPolicyBMTestCase.java

```diff
@@ -0,0 +1,108 @@
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.jca.core.connectionmanager.pool;
+
+import org.jboss.jca.core.api.connectionmanager.pool.PoolStatistics;
+import org.jboss.jca.core.connectionmanager.pool.mcp.ManagedConnectionPool;
+import org.jboss.jca.core.connectionmanager.rar.SimpleConnectionRequestInfoImpl;
+
+import javax.resource.spi.ConnectionRequestInfo;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.extension.byteman.api.BMRule;
+import org.jboss.arquillian.extension.byteman.api.BMRules;
+import org.jboss.shrinkwrap.api.spec.ResourceAdapterArchive;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * A PoolByCriNoTxDecrementCapacityTimedOutFIFOPolicyBMTestCase
+ * <p/>
+ * NOTE that this class is in org.jboss.jca.core.connectionmanager.pool and not in
+ * org.jboss.jca.core.connectionmanager.pool.strategy because it needs to access to
+ * AbstractPool's package protected methods.
+ * Please don't move it, and keep this class packaging consistent with AbstractPool's
+ *
+ * @author <a href="mailto:msimka@redhat.com">Martin Simka</a>
+ */
+@BMRules(value =
+      {
+            @BMRule(name = "start counters",
+                  targetClass = "PoolByCriNoTxDecrementCapacityTimedOutFIFOPolicyBMTestCase",
+                  targetMethod = "checkPool",
+                  action = "createCounter(\"counter1\", 2); createCounter(\"counter2\", 6)"),
+            @BMRule(name = "decrement counter1",
+                  targetClass = "AbstractConnectionListener",
+                  targetMethod = "isTimedOut",
+                  condition = "readCounter(\"counter1\")>0",
+                  action = "decrementCounter(\"counter1\"); decrementCounter(\"counter2\"); RETURN true;"),
+            @BMRule(name = "decrement counter2 - if 1 stop",
+                  targetClass = "AbstractConnectionListener",
+                  targetMethod = "isTimedOut",
+                  condition = "readCounter(\"counter1\")==0 && readCounter(\"counter2\")>2",
+                  action = "decrementCounter(\"counter2\")"),
+            @BMRule(name = "decrement counter2",
+                  targetClass = "AbstractConnectionListener",
+                  targetMethod = "isTimedOut",
+                  condition = "readCounter(\"counter2\")<3 && readCounter(\"counter2\")>0",
+                  action = "decrementCounter(\"counter2\"); RETURN true")
+      })
+public class PoolByCriNoTxDecrementCapacityTimedOutFIFOPolicyBMTestCase extends PoolByCriNoTxTestCaseAbstract
+{
+
+   /**
+    *
+    * deployment
+    *
+    * @return archive
+    */
+   @Deployment
+   public static ResourceAdapterArchive deployment()
+   {
+      return createNoTxDeployment(getCriIJWithDecrementer("TimedOutFIFODecrementer"));
+   }
+
+   @Override
+   public void checkPool() throws Exception
+   {
+      AbstractPool pool = getPool();
```

Comment:
Better description of the test case, since PoolByCri doesn't support prefil from a user PoV

Plus what the BM rules does